libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java

```diff
@@ -708,15 +708,16 @@ private void refreshAccessToken() throws IOException {
             // Then let's try to get a new authToken
             if (authTokenProvider != null) {
                 final String newAuthToken = authTokenProvider.getNewAuthToken();
-                if (newAuthToken != null) {
-                    setAuthToken(newAuthToken);
+
+				if (newAuthToken == null || authTokenProvider.getInstanceUrl() == null) {
+                    throw new RefreshTokenRevokedException("Could not refresh token");
```

Comment:
We need to let internal teams know to catch this exception now instead of `IOException`.